src/components/LeetcodeProfile.tsx

```diff
@@ -1,103 +1,107 @@
-import { useEffect, useState, useMemo } from 'react';
 import {
   Box,
   Link,
   Text,
-  VStack,
   Heading,
   Spinner,
-  Center,
+  Grid,
+  Flex,
   useColorModeValue,
+  IconButton,
+  Stat,
+  StatLabel,
+  StatNumber,
 } from '@chakra-ui/react';
-import { devPrint } from './utils/RandomUtils';
-import { LEETCODE_API_BASE_URL } from '../constants';
+import { FaExternalLinkAlt } from 'react-icons/fa';
 
-type LeetcodeProfileProps = {
+type LeetCodeStatsProps = {
   username: string;
+  stats: {
+    easy: number;
+    medium: number;
+    hard: number;
+  };
+  loading?: boolean;
```

Comment:
```suggestion
  loading: boolean;
```
I think since you provide a default value for `loading` in the component, you might not have to mark it as nullable in the type.